frontend/src/components/UserSearch.vue

```diff
@@ -208,6 +255,45 @@ export default {
         parameters += "&" + parameter + "=" + value;
       }
       return parameters;
+    },
+    getClientsAndRoles: function() {
+      return ClientsRepository.get()
+        .then(response => {
+          this.clients = response.data;
+          this.getClientRoles();
+        })
+        .catch(e => {
+          console.log(e);
+        });
+    },
+    getClientRoles: function() {
+      let getRolesRequests = this.clients
+        .map(client => ClientsRepository.getRoles(client.id));
+      Promise.all(getRolesRequests)
+        .then(roles => this.setClientRoles(roles));
+    },
+    setClientRoles: function(roles) {
+      let allRoles = roles
+        .flatMap(roles => roles.data)
+        .map(role => new ClientRole(role, this.clients));
+      allRoles.sort((a,b) => a.toString().localeCompare(b.toString()));
+      this.clientRoles.push(...allRoles);
+    },
```

Comment:
This is a suggestion, but I find it actually more confusing to have the logic spread out between `getClientsAndRoles`, `getClientRoles`, and `setClientRoles`. When you're using promise chaining and functional methods, I think it's easier to read it one continuous line. Basically replace all three of those methods with this one method:

```javascript
loadClientsAndRoles: function () {
  return ClientsRepository.get()
      .then(response => {
        this.clients = response.data;
        let roleRequests = this.clients.map(client => ClientsRepository.getRoles(client.id));
        Promise.all(roleRequests)
            .then(roles => {
              let allRoles = roles
                  .flatMap(roles => roles.data)
                  .map(role => new ClientRole(role, this.clients))
                  .sort((a, b) => a.toString().localeCompare(b.toString()));
              this.clientRoles.push(...allRoles);
            });
      })
      .catch(e => {
        console.log(e);
      });
},
```
If we were to keep these three methods, I would rename them. Naming them `get` and `set` brings to mind accessors, but they're not really accessors, they're initialization:

* `getClientsAndRoles` -> `loadClientsAndRoles`
* `getClientRoles` -> `loadClientRoles`
* `setClientRoles` -> `addRoles` maybe?